docs/receipt-verifier.md

```diff
@@ -0,0 +1,118 @@
+---
+id: receipt-verifier
+title: Receipt Verifier Service
+sidebar_label: Receipt verifier service
+---
+
+This page explains how to set up a receipt verifier service.
+
+As more creators adopt Web Monetization to serve premium and/or ad-free content, there’s a greater need for reliable payment verification. Web Monetization browser events, like `monetizationstart`, are helpful indicators of payment, but can be spoofed by savvy users to access exclusive content without paying.
+
+[Interledger STREAM receipts](https://interledger.org/rfcs/0039-stream-receipts/) provide payment recipients (such as web monetized site owners) with verifiable proofs of payment. As a payment recipient, you can verify STREAM receipts yourself by setting up your own verification service or by using a third-party service. You don't need to run your own Interledger connector or SPSP server to verify STREAM receipts.
+
+## Before you begin
+
+Your Web Monetization receiver (the digital entity receiving payments) must be set up to generate Interledger STREAM receipts. No action is required on your part. GateHub and Uphold already support receipts.
+
+## Service overview
+
+The receipt verifier service acts as a proxy between the Web Monetization sender and receiver.
+
+Before a payment is made, the WM sender sends a query to the WM receiver. The query passes through the verifier service, which adds a receipt secret and receipt nonce.
+
+When the WM receiver receives money from the payment, it generates a receipt (which includes a signature generated using the receipt secret) and sends the receipt to the WM sender.
+
+The WM sender passes the receipt to the verifier service so the verifier can confirm the payment (as only the WM receiver and verifier service know the receipt secret). The verifier must verify the receipt before accepting the amount as paid.
+
+## Install the receipt verifier service package
+
+You can get the [receipt verifier service package](https://github.com/coilhq/receipt-verifier) or run the verifier as a [Kubernetes service](https://github.com/coilhq/receipt-verifier/tree/main/config/base).
+
+**Example**
+
+This example shows Coil's implementation of the receipt verifier package, which requires Redis. You can write a different implementation that doesn't rely on Redis.
+
+```
+npm install
+npm run-script build
+sudo docker run -p 6379:6379 -d redis
+npm start
+```
+
+## Update your Web Monetization meta tag
+
+Typically, a Web Monetization meta tag looks something like this:
+
+`<meta name=“monetization" content="$wallet.example.com/alice">`
+
+For queries from the WM sender to be proxied by the receipt verifier service to your payment pointer:
+
+1. URI-encode your payment pointer. See [encodeURIComponent()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) on MDN Web Docs if you need help.
```

Comment:
will our site support MDX? https://v2.docusaurus.io/docs/markdown-features/#embedding-react-components-with-mdx I could make a little inline component for URI encoding pointers in the future.

Reply:
I believe the site will support that. The example block Docusaurus provides works locally, as do .mdx files. I also imported a component from Rebass (https://mdxjs.com/getting-started/#imports), and that was OK too.